cylc/flow/scripts/completion_server.py

```diff
@@ -459,21 +494,103 @@ def list_rel_dir(path: Path, base: Path) -> t.List[str]:
 
 
 async def list_flows(
-    _workflow: t.Optional[str],
     _partial: t.Optional[str]
 ) -> t.List[str]:
     """List values for the --flow option."""
     return ['all', 'none', 'new']
 
 
 async def list_colours(
-    _workflow: t.Optional[str],
     _partial: t.Optional[str]
 ) -> t.List[str]:
     """List values for the --color option."""
     return ['never', 'auto', 'always']
 
 
+async def list_outputs(
+    _partial: t.Optional[str],
+    tokens_list: t.Optional[t.List[Tokens]],
+):
+    """List task outputs."""
+    return (await _list_prereqs_and_outputs(tokens_list))[1]
+
+
+async def list_prereqs(
+    _partial: t.Optional[str],
+    tokens_list: t.Optional[t.List[Tokens]],
+):
+    """List task prerequisites."""
+    return (await _list_prereqs_and_outputs(tokens_list))[0] + ['all']
+
+
+async def _list_prereqs_and_outputs(
```

Comment:
Nice future feature might be to have whether each output is optional (include the `?`) on the autocomplete?

Reply:
You can't use the `?` symbol on the CLI, this is only for use in the graph.

Reply:
Pity we can't mark the required outputs off from the optional ones...